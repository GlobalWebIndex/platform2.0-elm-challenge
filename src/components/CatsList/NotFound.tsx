import { useContext } from "react";
import { Link } from "react-router-dom";
import { Grid, Typography } from "@mui/material";

import { CatContext } from "../../Context/CatContext";
import { sharedTextStyle } from "./styles";

const NotFound = () => {
  const { loading } = useContext(CatContext);

  const classes = sharedTextStyle();

  return !loading ? (
    <Grid item xs>
      <Link className={classes.link} to="/">
        <Typography className={classes.title} textAlign="center" variant="h2">
          Cat<span className={classes.loverColor}>Lover</span>
        </Typography>
      </Link>
      <Typography className={classes.textColor} textAlign="center" variant="h6">
        Unfortunately there are no cats to show
      </Typography>
    </Grid>
  ) : null;
};

export default NotFound;
