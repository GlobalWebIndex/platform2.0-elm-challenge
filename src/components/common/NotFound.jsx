import React from "react";
import { Link } from "react-router-dom";
import { useStyles } from "../App.styles";
import { Routes as RoutesConfig } from "../../constants";
import Header from "../common/Header";

const NotFound = () => {
  const classes = useStyles();
  return (
    <>
      <Header title="Whoa! This page does not exist!" />
      <div>
        Not sure how you ended up here but if you really love cats 🐈 and want
        to see more just{" "}
        <Link to={RoutesConfig.home} className={classes.contentLink}>
          go back to the homepage
        </Link>
      </div>
    </>
  );
};

export default NotFound;
