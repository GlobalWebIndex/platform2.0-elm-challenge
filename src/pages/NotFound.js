import React from 'react'
import notfound from '../img/notfound.jpg'
import {Container,Header,Image} from 'semantic-ui-react'
import { Link } from 'react-router-dom'
function NotFound() {
  return (
    <>
      <Container>
        <Header as='h1' textAlign='center'>Oops,invalid url.</Header>
            <div style={{marginTop:20,marginBottom:30,display: 'flex',alignItems: 'center' , justifyContent: 'center'}}>
              <Image style={{top:"+0px"}} size='big' fluid name='home'
                          as={Link} to="/"
                          src={notfound} />
            </div>
      </Container>
    </>
  )
}

export default NotFound