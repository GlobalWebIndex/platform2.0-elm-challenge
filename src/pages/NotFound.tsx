import React from "react";
import { Image, Button, Col, Row } from "react-bootstrap";
import { Link } from "react-router-dom";

const NotFound = () => (
  <React.Fragment>
    <Row>
      <Col className="text-center">
        <Image src="https://seranking.com/blog/wp-content/uploads/2021/01/404_01-min.jpg"></Image>
      </Col>
    </Row>
    <Row>
      <Col className="text-center mt-1">
        <Link to="/">
          <Button variant="dark">Go home</Button>
        </Link>
      </Col>
    </Row>
  </React.Fragment>
);

export default NotFound;
