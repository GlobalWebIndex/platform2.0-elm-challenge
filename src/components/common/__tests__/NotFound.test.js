import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import NotFound from "../NotFound";

describe("<NotFound/>", () => {
  it("renders", () => {
    const { container } = render(<NotFound />, { wrapper: BrowserRouter });
    expect(container).toMatchSnapshot();
  });
});
