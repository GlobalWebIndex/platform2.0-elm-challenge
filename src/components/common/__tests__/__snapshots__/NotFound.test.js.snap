// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound/> renders 1`] = `
<div>
  <header
    class="title"
    data-testid="header"
  >
    <h1>
      Whoa! This page does not exist!
    </h1>
  </header>
  <div>
    Not sure how you ended up here but if you really love cats 🐈 and want to see more just
     
    <a
      class="contentLink-0-2-2"
      href="/"
    >
      go back to the homepage
    </a>
  </div>
</div>
`;
